### PR TITLE
Provides additional guidance to developers, and ensures we can build/test/iterate locally, against skatepark, and against cloud

### PR DESCRIPTION
**Pull request by SavagePencil**

### CODE REVIEW

File: example-client/.env.client.example [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copy this file to a new file named '.env' to automatically load these
+# values as environment variables when running the client.
+# You will need to fill in the values based on which scenario you are
+# running against:
+# 1. A client testing against a Structure registered locally in the
```
Comment (shhlife) [inline example-client/.env.client.example]: it maybe obvious - but would it be useful to explain what a "client" is?

File: example-client/.env.client.example [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# If the Structure is running in the Skatepark emulator, this value
+# will be the ID returned when building or registering the Structure
+# from the CLI.
+# In Griptape Cloud, it will be the ID of the structure as found on
+# https://cloud.griptape.ai/structures .
+GT_STRUCTURE_ID=YOUR_STRUCTURE_ID_HERE
+
+# If running against the Skatepark emulator, you will need to set
+# this environment variable (or put it in a .env file) to tell the
+# client which URL Skatepark is listening on. The default value is
```
Comment (shhlife) [inline example-client/.env.client.example]: does this mean if they're happy with the default, they don't need to change it?

File: example-client/client.py [hunk 3]
```diff
@@ -10,10 +13,30 @@
...
+# of the Structure that you registered with Skatepark. For a Structure
+# hosted in Griptape Cloud, it will be the ID of the Structure you 
+# are running, which can be found at https://cloud.griptape.ai/structures 
 GT_STRUCTURE_ID = os.environ["GT_STRUCTURE_ID"]
 
+# The Skatepark emulator does NOT require a Griptape API key.
+# However, Griptape Cloud requires a valid API Key in order
+# to authorize calling the API. You can generate a key by
+# visiting https://cloud.griptape.ai/keys
+GT_API_KEY = os.environ.get("GT_CLOUD_API_KEY", "GRIPTAPE API KEY ONLY NEEDED FOR STRUCTURES IN GRIPTAPE CLOUD")
```
Comment (shhlife) [inline example-client/client.py]: ooo fancy!

File: example-client/client.py [hunk 4]
```diff
@@ -24,26 +47,44 @@ def run_structure(input: str) -> Optional[str]:
     Returns:
         The output of the Structure program.
     """
-    # Start a run with the args "What is 5 ^ 2".
+    # Start a run with the input args.
```
Comment (shhlife) [inline example-client/client.py]: do we show an example anywhere of what the input args should look like?

Comment (shhlife) [reply example-client/client.py]: I mean, I see them down below - in the run - but wondering if we want to explain in more detail what they can be

File: .env.structure.example [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Copy this file to a new file named '.env' to automatically load these
+# values as environment variables when running the Structure.
+
+# The Agent example included expects an OpenAI key. Supply yours here:
+OPENAI_API_KEY=YOUR_KEY_HERE
+
+# The event driver expects a Griptape API Key as a parameter in order to emit events.
```
Comment (collindutter) [inline .env.structure.example]: `as a parameter in order to emit events` -> `as a parameter in order to emit events back to Griptape Cloud`

Comment (collindutter) [inline .env.structure.example]: `Griptape API Key` -> `Griptape Cloud Api Key`

File: example-client/.env.client.example (hunk 1 above)
Comment (collindutter) [reply example-client/.env.client.example]: Maybe specify `client program`?

File: example-client/.env.client.example [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+# When running in Griptape Cloud, the Structure first reports
+# a status of QUEUED before transitioning to RUNNING. Skatepark
+# emulates this behavior by applying a default time to remain 
+# in the QUEUED state in order to allow your client to 
+# handle this state appropriately.
+# You can override the time that the Structure remains in
+# the QUEUED state within Skatepark by setting the environment
+# variable GT_SKATEPARK_QUEUE_DELAY to the desired time in seconds.
+# GT_SKATEPARK_QUEUE_DELAY=VALUE_IN_SECONDS_TO_QUEUE
```
Comment (collindutter) [inline example-client/.env.client.example]: I think we should provide a valid default, even in the example. I.e. `GT_SKATEPARK_QUEUE_DELAY=2`

File: example-client/.env.client.example (hunk 2 above)
Comment (collindutter) [reply example-client/.env.client.example]: When users point their client to Cloud they'll need to update this value to `https://cloud.griptape.ai`

File: example-client/client.py (hunk 3 above)
Comment (collindutter) [inline example-client/client.py]: With the `.env` examples do we need to provide a fallback value here? I think I'd rather it error out indicating that the value is missing and their `.env` configuration is wrong.

File: example-client/client.py [hunk 7]
```diff
@@ -24,26 +47,44 @@ def run_structure(input: str) -> Optional[str]:
...
     # These args will be passed into our Structure program as standard input.
-    structure_run = create_structure_run(GT_STRUCTURE_ID, {}, [input])
+    structure_run = create_structure_run(
+        host=HOST,
+        api_key=GT_API_KEY,
+        structure_id=GT_STRUCTURE_ID, 
+        env={}, 
+        args=[input]
+        )
```
Comment (collindutter) [inline example-client/client.py]: Kind of strange formatting, can you run this through a VsCode format?

**Review by collindutter — CHANGES_REQUESTED** [review]

File: example-client/.env.client.example (hunk 1 above)
Comment (SavagePencil) [reply example-client/.env.client.example]: check latest

File: example-client/.env.client.example (hunk 2 above)
Comment (SavagePencil) [reply example-client/.env.client.example]: still an issue?

File: example-client/client.py (hunk 4 above)
Comment (SavagePencil) [reply example-client/client.py]: verify latest

File: .env.structure.example (hunk 5 above)
Comment (SavagePencil) [reply .env.structure.example]: think I got it

Comment (SavagePencil) [reply .env.structure.example]: did a consistency pass; good call

File: example-client/.env.client.example (hunk 6 above)
Comment (SavagePencil) [reply example-client/.env.client.example]: bleh I don't agree but I don't disagree strongly enough to care. Updated.

File: example-client/client.py (hunk 7 above)
Comment (SavagePencil) [reply example-client/client.py]: something was FUBAR with Black

File: example-client/client.py (hunk 3 above)
Comment (SavagePencil) [reply example-client/client.py]: We do warn the user if they don't specify one. And Jason thought it was "fancy." I could go either way.

**Review by collindutter — APPROVED** [review]

**Pull request merged**